engine/admin/start-containers-automatically.md

```diff
@@ -0,0 +1,57 @@
+---
+description: How to generate scripts for upstart, systemd, etc.
+keywords: systemd, upstart, supervisor, docker, documentation,  host integration
+redirect_from:
+- /engine/articles/host_integration/
+- /engine/admin/host_integration/
+title: Start containers automatically
+---
+
+Docker provides [restart policies](/engine/reference/run.md#restart-policies---restart)
+to control whether your containers start automatically when they exit, or when
+Docker restarts. Restart policies ensure that linked containers are started in
+the correct order. Docker recommends that you use restart policies, and avoid
+using process managers to start containers.
+
+## Use a restart policy
+
+To configure the restart policy for a container, use the `--restart` flag
+when using the `docker run` command. The value of the `--restart` flag can be
+any of the following:
+
+| Flag             | Description                                                                                     |
+|:-----------------|:------------------------------------------------------------------------------------------------|
+| `no`             | Do not automatically restart the container. (the default)                                       |
+| `on-failure`     | Restart the container if it exits due to an error.                                              |
+| `unless-stopped` | Restart the container unless it is explicitly stopped or Docker itself is stopped or restarted. |
+| `always`         | Always restart the container if it stops.                                                       |
+
+The following example starts a Redis container and configures it to always
+restart unless it is explicitly stopped or Docker is restarted.
+
+```bash
+$ docker run -dit --restart unless-stopped redis
+```
+
```

Comment:
Some "pitfalls" when using restart policies (not sure they should be mentioned in this context, but writing down FYI);

- the restart policy takes effect after the container is started successfully (i.e., after the container is running 10 seconds or more, monitoring is started)
- this means that if a container _fails to start_, it is not re-attempted to start
- of course; this is different with _services_ because then swarm kit will keep retrying to start new tasks
- restart policies do _not_ take effect if you manually stop a container; in that case, the container is not restarted, however it will be restarted when the daemon is restarted. The container is not restarted, because manually stopping a container is a deliberate choice by the user.

Reply:
@thaJeztah , per your 4th bullet point, should @mistyhacks 's text on `unless-stopped` include a 2nd `explicitly`? or should the part about `Docker` be removed entirely? 

> Restart the container unless it is explicitly stopped or Docker itself is _explicitly_ stopped or restarted.

Reply:
I ask because of #6287

Reply:
Yes, I think that PR is correct (there have been some corner-cases though, so I guess the behaviour wasn't fully defined for all of those)

Reply:
I see. Thank you.